### PR TITLE
Initial v0.19 docs

### PR DESCRIPTION
**Pull request by jtcohen6**
## Description & motivation
- Add migration guide for v0.19.0
- Expanded docs for dbt artifacts
    - There's still some WIP in dbt v0.19 around run results + sources
    - We should actually host the JSON schemas at schemas.getdbt.com/dbt/ before merging this PR or releasing v0.19.0
- State
    - Draft and centralize some prose about WTH `--state` is. I have it nested under `artifacts/` right now, it could go elsewhere.
    - Write up major known caveats about `state:modified`. This feels like the Big Thing we need before we can take Slim CI out of beta.
- Little things
    - Update documented RPC commands with their actual accepted parameters (`state` and `defer` are new in v0.19, the rest were just out of date/sync)
    - Added to docs stub for `invocation_id`
    - Add details about new BQ connection method, `oauth-secrets`

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [x] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [ ] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

## Checklist
If you added new pages (delete if not applicable):
- [x] The page has been added to `website/sidebars.js`
- [x] The new page has a unique filename

### CODE REVIEW

File: _redirects [hunk 1]
```diff
@@ -248,6 +248,7 @@
 /docs/writing-code-in-dbt/macros	/docs/building-a-dbt-project/jinja-macros	302
 /docs/writing-code-in-dbt/using-jinja	/tutorial/using-jinja	302
 /faqs/getting-help/	/docs/guides/getting-help/	302
+/reference/dbt-artifacts	/reference/artifacts/dbt-artifacts	302
```
Comment (clrcrl) [inline _redirects:251]: so, uh, we are now keeping these alpha-ordered :)

File: website/docs/docs/guides/migration-guide/upgrading-to-0-19-0.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+---
+title: "Upgrading to 0.19.0"
+
+---
+
+:::info Prerelease
+
+dbt v0.19.0 is currently in beta. Please post in dbt Slack #prereleases with questions or issues.
```
Comment (clrcrl) [inline website/docs/docs/guides/migration-guide/upgrading-to-0-19-0.md]:
```suggestion
dbt v0.19.0 is currently in beta. Please post in dbt Slack #dbt-prereleases with questions or issues.
```
(They'll still find their way there if they search "prereleases"

File: _redirects (hunk 1 above)
Comment (clrcrl) [inline _redirects]: We're now trying to keep these alpha-ordered. We should add some simple CI on this or something I think

File: website/docs/docs/guides/migration-guide/upgrading-to-0-19-0.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+## New features
+
+### Artifacts
+- The JSON artifacts produced by dbt—manifest, catalog, run results, and sources—are simpler to consume and more clearly documented.
+
+### State
+- dbt is a little bit smarter at identifying `state:modified` "false positives" that previously resulted from env-based config in `dbt_project`.
+
+### Snapshots
+- If the config `invalidate_hard_deletes` is enabled, `dbt snapshot` will update records whose unique key no longer exist in the snapshot query. Should those uniquely identified records "revive," `dbt snapshot` will re-add them.
```
Comment (clrcrl) [inline website/docs/docs/guides/migration-guide/upgrading-to-0-19-0.md:32]: dope!

File: website/docs/reference/artifacts/catalog.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+---
+title: Catalog
+---
+
+### catalog.json
```
Comment (clrcrl) [inline website/docs/reference/artifacts/catalog.md]: Future art: do we have the jsonschema for this? Would be cool to render that in these docs :)

File: website/docs/reference/artifacts/catalog.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+_Produced by:_ `dbt docs generate`
+
+This file contains information from your data warehouse about the tables and views produced by the models in your project. dbt uses this file to render information like column types and table statistics into the docs site.
```
Comment (clrcrl) [inline website/docs/reference/artifacts/catalog.md]: Should this also mention sources?

File: website/docs/reference/artifacts/dbt-artifacts.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+All artifacts produced by dbt include a `metadata` dictionary.
+
+- `dbt_version`: Version of dbt that produced this artifact.
+- `dbt_schema_version`: URL of this artifact's schema. See notes below.
+- [`invocation_id`](invocation_id)
+- `generated_at`: Timestamp in UTC when this artifact was produced.
+- `env`: Any environment variables prefixed with `DBT_ENV_CUSTOM_ENV_` will be included in a dictionary, with the prefix-stripped variable name as its key.
+
+#### Notes:
+- The structure of dbt artifacts is canonized by [JSON schemas](https://json-schema.org/) hosted at https://schemas.getdbt.com/dbt/
```
Comment (clrcrl) [inline website/docs/reference/artifacts/dbt-artifacts.md]: The https://schemas.getdbt.com/dbt/ link doesn't work — is that expected?

File: website/docs/reference/artifacts/run-results.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+---
+
+_Produced by:_
+- `dbt run`
+- `dbt test`
+- `dbt seed`
+- `dbt snapshot`
+- `dbt docs generate`
+
+This file contains information about the last run of dbt. Crucially, it contains the compiled SQL for every model in your project. dbt uses this file to render the compiled sql directly into the documentation site.
```
Comment (clrcrl) [inline website/docs/reference/artifacts/run-results.md]:
```suggestion
This file contains information about the last run of dbt. Crucially, it contains the compiled SQL for every model in your project. dbt uses this file to render the compiled SQL directly into the documentation site.
```

File: website/docs/reference/artifacts/state.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
+---
+title: "State"
```
Comment (clrcrl) [inline website/docs/reference/artifacts/state.md]: Hmm I'm not sure how this fits into "artifacts"

File: website/docs/reference/node-selection/state-comparison-caveats.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
+---
+title: "Caveats to state comparison"
```
Comment (clrcrl) [inline website/docs/reference/node-selection/state-comparison-caveats.md:2]: Hmmm I don't feel like I'm in a good spot to review this doc — I feel a little far-removed from the nuances of this. Going to give it a 👍 but know that I didn't interrogate this too closely 😅

File: website/docs/reference/node-selection/state-comparison-caveats.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+State comparison works by identifying discrepancies between two manifests.  Those discrepancies could be the result of:
+
+1. Changes made to a project in development
+2. Env-aware logic that causes different behavior based on the `target`, env vars, etc.
+
+dbt will do its best to capture *only* changes that are the result of modifications made in development. In projects with intricate env-aware logic, dbt will err on the side of running too many models (i.e. false positives). Over the next several versions of dbt, We're working on:
+- iterative improvements to dbt's built-in dectective abilities
+- better options for more complex projects, in the form of more-specific subselectors (see [this issue](https://github.com/fishtown-analytics/dbt/issues/2704))
+
+<Changelog>
```
Comment (clrcrl) [inline website/docs/reference/node-selection/state-comparison-caveats.md:60]: This is dumb, but you need a new line to render this as bullet points
```suggestion
<Changelog>

```

File: website/docs/reference/node-selection/state-comparison-caveats.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+- v0.18.0: All env-aware logic results in false positives during state comparison,
+when comparing against a manifest generated with a different target.
+- v0.19.0: dbt can detect
```
Comment (clrcrl) [inline website/docs/reference/node-selection/state-comparison-caveats.md]: what can it detect :eyes:

File: website/docs/reference/warehouse-profiles/bigquery-profile.md [hunk 12]
```diff
@@ -6,16 +6,16 @@ title: "BigQuery Profile"
 
 BigQuery targets can be specified using one of three methods:
 
-1. oauth
+1. oauth via `gcloud`
+2. oauth via access token(s)
```
Comment (clrcrl) [inline website/docs/reference/warehouse-profiles/bigquery-profile.md]: The list here is `1, 2, 2, 3`. This ends up rendering correctly, but would be good to fix :)

File: website/docs/reference/warehouse-profiles/bigquery-profile.md [hunk 13]
```diff
@@ -6,16 +6,16 @@ title: "BigQuery Profile"
 
 BigQuery targets can be specified using one of three methods:
```
Comment (clrcrl) [inline website/docs/reference/warehouse-profiles/bigquery-profile.md]:
```suggestion
BigQuery targets can be specified using one of four methods:
```

File: website/docs/reference/warehouse-profiles/bigquery-profile.md [hunk 14]
```diff
@@ -6,16 +6,16 @@ title: "BigQuery Profile"
...
 2. a service account file
 3. service account json
 
 For local development, we recommend using the oauth method. If you're scheduling dbt on a server, you should use the service account auth method instead.
 
 BigQuery targets should be set up using the following configuration in your `profiles.yml` file.
 
-### oauth Authentication
-
+### OAuth Local Authentication
```
Comment (clrcrl) [inline website/docs/reference/warehouse-profiles/bigquery-profile.md]: Can we make these headings match the above list? It's a little easier to navigate when they are consistent

File: website/docs/reference/warehouse-profiles/bigquery-profile.md [hunk 15]
```diff
@@ -103,11 +173,11 @@ my-bigquery-db:
 
 </File>
 
-## Oauth Authorization
 
-To connect to BigQuery using the `oauth` method, follow these steps:
 
+## OAuth Authorization (local)
```
Comment (clrcrl) [inline website/docs/reference/warehouse-profiles/bigquery-profile.md]: Is this the same as `### OAuth Local Authentication`

File: website/sidebars.js [hunk 16]
```diff
@@ -76,6 +76,7 @@ module.exports = {
             "docs/guides/migration-guide/upgrading-to-0-16-0",
             "docs/guides/migration-guide/upgrading-to-0-17-0",
             "docs/guides/migration-guide/upgrading-to-0-18-0",
+            "docs/guides/migration-guide/upgrading-to-0-19-0",
```
Comment (clrcrl) [inline website/sidebars.js:71]: how do you feel about reverse-ordering these?

**Review by clrcrl — APPROVED** [review]
A few different comments, but giving it a 👍  under the assumption that they get addressed :)

File: website/docs/reference/artifacts/catalog.md (hunk 4 above)
Comment (jtcohen6) [reply website/docs/reference/artifacts/catalog.md]: agree!!

File: website/docs/reference/artifacts/dbt-artifacts.md (hunk 6 above)
Comment (jtcohen6) [reply website/docs/reference/artifacts/dbt-artifacts.md]: yeah... that's TK. I'll add a note.

File: website/docs/reference/artifacts/run-results.md (hunk 7 above)
Comment (jtcohen6) [reply website/docs/reference/artifacts/run-results.md]: heh... this was an artifact of the previous artifact documentation

File: website/docs/reference/artifacts/state.md (hunk 8 above)
Comment (jtcohen6) [reply website/docs/reference/artifacts/state.md]: yeah, fair. I'll just have it live by itself for now

File: website/docs/reference/node-selection/state-comparison-caveats.md (hunk 11 above)
Comment (jtcohen6) [reply website/docs/reference/node-selection/state-comparison-caveats.md]: lol thanks for catching

File: website/docs/reference/warehouse-profiles/bigquery-profile.md (hunk 15 above)
Comment (jtcohen6) [reply website/docs/reference/warehouse-profiles/bigquery-profile.md]: renamed to distinguish

File: website/sidebars.js (hunk 16 above)
Comment (jtcohen6) [reply website/sidebars.js:71]: i love it!

**Comment by jtcohen6** [comment]
@clrcrl Thank you so much for the detailed review! Really, really, really helpful

**Pull request merged**